pkg/virtctl/vm/vm_test.go

```diff
@@ -20,44 +19,67 @@ var _ = Describe("VirtualMachine", func() {
 	const vmName = "testvm"
 	var vmInterface *kubecli.MockVirtualMachineInterface
 	var ctrl *gomock.Controller
-	vm := kubecli.NewMinimalVM("testvm")
 
 	BeforeEach(func() {
 		ctrl = gomock.NewController(GinkgoT())
 		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
 		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+
 		// create mock interface
 		vmInterface = kubecli.NewMockVirtualMachineInterface(ctrl)
-		// set up mock client behavior
-		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).AnyTimes()
-		// set up mock interface behavior
-		vmInterface.EXPECT().Get(vm.Name, gomock.Any()).Return(vm, nil).AnyTimes()
-		vmInterface.EXPECT().Get(unknownVM, gomock.Any()).Return(nil, errors.New("unknown VM")).AnyTimes()
-		vmInterface.EXPECT().Patch(vmName, gomock.Any(), gomock.Any()).Return(vm, nil).AnyTimes()
-		vmInterface.EXPECT().Restart(vmName).Return(nil).AnyTimes()
-		vmInterface.EXPECT().Restart(unknownVM).Return(errors.New("unknown VM")).AnyTimes()
+
 	})
 
 	Context("should patch VM", func() {
 		It("with spec:running:true", func() {
-			buffer := &bytes.Buffer{}
+			vm := kubecli.NewMinimalVM(vmName)
```

Comment:
Don't we still want to read the buffer to ensure the output is as expected, even if that output is ""

Reply:
I do not think so. The test I have written is equivalent to previous one. I make sure every function is called with the proper number of times and with proper parameters. The output buffer was, imho used only as a helper for the test. Previously the mocking was done in a generic way,  allowing unlimited calls to every function used in the test.